ap/aputils/trainee_utils.py

```
@@ -1,13 +1,19 @@
 from accounts.models import Trainee
 
 def is_trainee(user):
-    t = user.type
+    try:
```

AnonymousUser which you get if you're not logged in doesn't have a type. I removed the try/catch and just handle this the same way as `exams/context_processor.py` used to, just check if there's a type attr on user. I'm not sure if it's better, maybe what we should do is have a more explicit exception to catch so that we don't have the silent errors.